src/mutable-prim-array-types.hs

```diff
@@ -31,74 +35,106 @@ unsafeCopy ::
     -> Int -- ^ offset into source array
     -> Int -- ^ number of elements to copy
     -> m ()
-unsafeCopy (Array dst#) (I# doff#) (Array src#) (I# soff#) (I# n#)
-  = primitive_ (copyMutableByteArray#
-      src#
-      (soff# *# (sizeOf# (undefined :: a)))
-      dst#
-      (doff# *# (sizeOf# (undefined :: a)))
-      (n# *# (sizeOf# (undefined :: a)))
-    )
+unsafeCopy (Array dst#) (I# doff#) (Array src#) (I# soff#) (I# n#) =
+    let toBytes cnt# = cnt# *# (sizeOf# (undefined :: a))
+     in primitive_ $
+            copyMutableByteArray#
+                src# (toBytes soff#) dst# (toBytes doff#) (toBytes n#)
 
+-------------------------------------------------------------------------------
+-- Length
+-------------------------------------------------------------------------------
+
+-- XXX rename to byteCount?
+{-# INLINE byteLength #-}
+byteLength :: Array s a -> Int
+byteLength (Array arr#) = I# (sizeofMutableByteArray# arr#)
+
+-- XXX Rename length to elemCount so that there is no confusion bout what it
```

Comment:
I don't think there would be any confusion with `length` but `elemCount` sounds to the point.

Reply:
`length` is used everywhere, in lists, streams etc, so that way it is consistent. Maybe we can just leave it like that.